### PR TITLE
CM-45: Managed vs Unmanaged code (Garbage Collection)

### PR DESCRIPTION
**Pull request by lasaetarubia**

### CODE REVIEW

File: csharp-advanced-topics/ManagedVsUnmanagedCodeInCSharp/Tests/Test.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        public void WhenAllocateMemoryCalled_ThenAllocateMemory()
+        {
+            var memoryManager = new ManagedMemoryManager();
+
+            var memoryBeforeAllocation = GC.GetTotalMemory(false);
+            memoryManager.AllocateMemory();
+            var memoryAfterAllocation = GC.GetTotalMemory(false);
+
+            Assert.True(memoryBeforeAllocation < memoryAfterAllocation);
+
```
Comment (PhilBroderick) [inline csharp-advanced-topics/ManagedVsUnmanagedCodeInCSharp/Tests/Test.cs:17]: Empty line here that should be removed.

**Review by PhilBroderick — CHANGES_REQUESTED** [review]
Looks fine. Small formatting change. I'll get in touch with the team about the build errors.

File: csharp-advanced-topics/ManagedVsUnmanagedCodeInCSharp/ManagedVsUnmanagedCodeInCSharp/ManagedMemoryManager.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        private readonly List<int> _numbers;
+
+        public ManagedMemoryManager()
+        {
+            _numbers = new List<int>();
+        }
+
+        public void AllocateMemory()
+        {
+            for (var i = 0; i < 100000; i++)
```
Comment (PhilBroderick) [inline csharp-advanced-topics/ManagedVsUnmanagedCodeInCSharp/ManagedVsUnmanagedCodeInCSharp/ManagedMemoryManager.cs:14]: In the article, we are using 1,000,000, whereas here it is 10,000.

**Review by PhilBroderick — CHANGES_REQUESTED** [review]
All the C++ code should be removed. Everything else looks fine.

**Comment by lasaetarubia** [comment]
retest this please

**Review by PhilBroderick — APPROVED** [review]

**Pull request merged**